modin/core/dataframe/pandas/partitioning/partition_manager.py

```diff
@@ -335,51 +335,53 @@ def broadcast_apply(cls, axis, apply_func, left, right, other_name="r"):
             Axis to apply and broadcast over.
         apply_func : callable
             Function to apply.
-        left : NumPy 2D array
-            Left partitions.
-        right : NumPy 2D array
-            Right partitions.
+        left : np.ndarray
+            NumPy array of left partitions.
+        right : np.ndarray
+            NumPy array of right partitions.
         other_name : str, default: "r"
             Name of key-value argument for `apply_func` that
             is used to pass `right` to `apply_func`.
 
         Returns
         -------
-        NumPy array
-            An of partition objects.
+        np.ndarray
+            NumPy array of result partition objects.
 
         Notes
         -----
         This will often be overridden by implementations. It materializes the
         entire partitions of the right and applies them to the left through `apply`.
         """
+        # This check will no longer be necessary here if we transfer this responsibility
+        # to, for example, the `axis_partition` constructor everywhere.
+        # On the other hand, it looks like a better solution here is to make this function
+        # internal, so that the partition itself calls this function when necessary.
         [obj.drain_call_queue() for row in right for obj in row]
```

Comment:
Let's allow axis partitions of the right object to drain the call queues in the constructor. This responsibility should lay on them.

Reply:
shouldn't this be handled in a separate PR? I wonder however why this line wasn't present in engine-specific implementations...